### DISCUSSION TITLE
Is it possible to get the AST objects in Python?

### DISCUSSION THREAD

**Discussion by r-bar** [inducer/starlark-pyo3#33] [Ideas]
I would like to be able to run my own transformations on the AST, but it appears the AST object is opaque. Is there a way to expose a tree of python objects from a parsed Ast object so I can traverse it in python?

**Comment by inducer** [comment]
AFAICT, the AST is not part of the documented interface of starlark-rust: https://docs.rs/starlark/latest/starlark/. You could try and petition them to document it, then making a wrapper for that functionality might be feasible.
